src/main.c

```diff
@@ -49,6 +52,61 @@ static void softReset_button_callback(struct device *port, struct gpio_callback
 	soft_reboot();
 }
 
+
+void cdc_acm_descriptor_callback (cdc_acm_cfg_t *cfg)
+{
+	customer_data_t *otp_data = (customer_data_t *)CUSTOMER_DATA_ADDRESS;
+
+	memset(cfg->vendor_string, '\0', 32);
+	memset(cfg->product_string, '\0', 32);
+	memset(cfg->serial_string, '\0', 32);
+
+	//read the USB descriptors from OTP and use some default values if it has not been properly populated
+	if((otp_data->vendor_name_len>0) && (otp_data->vendor_name_len<=32))
+	{
+		memcpy(cfg->vendor_string, otp_data->vendor_name, otp_data->vendor_name_len);
+	}
+	else
+	{
+		char *vendor = "Intel";
```

Comment:
Note that string literals go in the .data segment-- only the *pointer* (i.e. `vendor`) will be allocated on the stack. This is why it's generally best practice to put string literals at the top of the file, and make them explicitly `const`.

However, if you want to force them to go on the stack, you can use the same literals to initialize an array of known size, e.g.

    char vendor[vendor_len] = "Intel";